server/routerlicious/packages/routerlicious-base/src/alfred/routes/api/deltas.ts

```diff
@@ -92,7 +98,8 @@ export function create(
         "/:tenantId/:id",
         validateRequestParams("tenantId", "id"),
         verifyStorageToken(tenantManager, config),
-        throttle(throttler, winston, commonThrottleOptions),
+        throttle(clusterThrottlers.get(Constants.alfredRestThrottleIdGetDeltas), winston, getDeltasThrottleOptions),
```

Comment:
I'm wondering if per-cluster throttling should happen before token verification. On the one hand, it would protect against DOS without a valid token which would hit our auth DB enough to cause problems, but on the other hand, it would allow someone to send enough invalid requests that they cause everyone else to be throttled which would effectively accomplish the same task.

@yangg-msft what do you think?

Reply:
qq: Would the validation calls hit the auth db or redis?

Reply:
@pradeepvairamani, the answer is both. Kara has cached the tenant keys in Redis, but if the validation fails with both keys, we'd fetch the keys from the group DB, so a malicious user can still choke us. Due to this reason, I'd suggest to throttle before validation.

Reply:
I think we should throttler before token verification.

With the first request, the key will be stored in redis cache and subsequent calls it will be fetched from the cache to validate the token in riddler.

Reply:
oops. yang's message didn't show up while i was reviewing. Glad we agree. :)

Reply:
Sounds good! Made the change.